libcaf_core/tests/legacy/mixin/requester.cpp

```diff
@@ -363,4 +363,23 @@ TEST_CASE("GH-1299 regression blocking") {
   }
 }
 
+TEST_CASE("GH-698 regression responding to a request drops the strong "
+          "reference count") {
+  auto weak_client_ref = [this] {
+    return actor_cast<weak_actor_ptr>(
+      sys.spawn([this](event_based_actor* self) {
```

Comment:
This is not what I would expect here. By only having a weak ref to this actor, we will terminate it as unreachable once it processes the response message.

The reported bug was about *not* terminating the actor. The fixed version must work with just:

```suggestion
  auto client = sys.spawn([this](event_based_actor* self) {
```

Reply:
I guess I can keep the client in scope and expect the ref count to drop to 1.